test/models/parser/twitter_profile_test.rb

```diff
@@ -71,62 +60,123 @@ def empty_doc
     assert_equal true, match_four.is_a?(Parser::TwitterProfile)
   end
 
-  test "should send Twitter error to Errbit and return default values" do
-    Twitter::REST::Client.any_instance.stubs(:user).raises(Twitter::Error)
+  test "it makes a get request to the user lookup by username endpoint successfully" do
+    stub_configs({'twitter_bearer_token' => 'test' })
+    
+    WebMock.stub_request(:get, "https://api.twitter.com/2/users/by")
+      .with(query: query)
+      .with(headers: { "Authorization": "Bearer test" })
+      .to_return(status: 200, body: response_fixture_from_file('twitter-profile-response-success.json'))
+
+    data = Parser::TwitterProfile.new('https://m.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_equal '2009-04-07T15:40:56.000Z', data['published_at']
+    assert_equal '@fake_user', data['username']
+  end
+
+  test "it makes a get request to the user lookup by username endpoint and raises an error when 401 is returned" do
+    stub_configs({'twitter_bearer_token' => 'test' })
+
+    WebMock.stub_request(:get, "https://api.twitter.com/2/users/by")
+      .with(query: query)
+      .with(headers: { "Authorization": "Bearer test" })
+      .to_return(status: 401)
+
+    assert_raises ProviderTwitter::ApiError do
```

Comment:
this is a really well-written test, but I think we want this to *not* raise the error, since we expect our parsers to always return an object and handle their internal errors themselves. this is why we normally wrap parse_data internals in an `handle_exceptions(StandardError)` block. I think that could be a good approach here, too! instead, we should do something like the test below where we verify the exception is sent to Sentry